### PR TITLE
Feature/sample right

### PR DESCRIPTION
**Pull request by sandorbanvitagroup**

### CODE REVIEW

File: src/app/_constants/api.ts [hunk 1]
```diff
@@ -14,6 +14,24 @@ export const API_ROUTE_PERSONS = {
...
+
+export const API_ROUTE_ADDITIONAL_TESTS = {
+  ENDPOINT: 'additionaltests',
+  GET_ALL: 'additionaltests/indexList',
+  UPDATE: 'additionaltests/update',
+  ADD: 'additionaltests/add',
+  DELETE: 'additionaltests/delete',
+  GET_BY_SAMPLE_ID: 'additionaltests/query/samples',
+};
+
```
Comment (adriandumavitagroup) [inline src/app/_constants/api.ts:34]: put a comment for the endpoints that are hardcoded : // toDo: hardcoded endpoint. Needs backend

File: src/app/_services/api/base.service.ts [hunk 2]
```diff
@@ -89,6 +89,18 @@ export class BaseService<T extends Resource> {
...
+    let endpoint = this.endpoint.ENDPOINT;
+    if (this.endpoint.GET_BY_SAMPLE_ID) {
+      endpoint = this.endpoint.GET_BY_SAMPLE_ID;
+    }
+
+    return this.httpClient
+      .post(`${this.helperService.getApiUrl()}/${endpoint}`, [id])
+      .pipe(map((data: any) => this.serializer.fromJson(data) as T));
+  }
+
```
Comment (adriandumavitagroup) [inline src/app/_services/api/base.service.ts:104]: this is not correct. We have a method getById in baseService. Why not use that?

File: src/app/shared/edge-panel/edge-panel.component.ts [hunk 3]
```diff
@@ -48,18 +49,32 @@ export class EdgePanelComponent implements OnInit, OnDestroy {
...
-        next: (response: any) => {
-          this.items = response.elements;
-        },
-        error: (err: any) => {
-          this.notificationService.error(err);
-        },
-        complete: () => {},
-      })
-    );
+    console.log('ssssss', this.config);
```
Comment (adriandumavitagroup) [inline src/app/shared/edge-panel/edge-panel.component.ts]: remove console log

File: src/app/_services/api/base.service.ts (hunk 2 above)
Comment (sandorbanvitagroup) [reply src/app/_services/api/base.service.ts:104]: I did a separate one, as this call is a post and not a get as getById is. Also we need to pass the sampleId in the body and not in the url. So did it like this to be clear that this is another kind of call. We could add callType parameter to specify if it is post or get, but I'm not sure we want to do that. Like this it's obvious that it has nothing to do with getById.

File: src/app/_constants/api.ts (hunk 1 above)
Comment (sandorbanvitagroup) [reply src/app/_constants/api.ts:34]: Done.

File: src/app/shared/edge-panel/edge-panel.component.ts (hunk 3 above)
Comment (sandorbanvitagroup) [reply src/app/shared/edge-panel/edge-panel.component.ts]: Done.

File: src/app/_services/api/base.service.ts (hunk 2 above)
Comment (adriandumavitagroup) [reply src/app/_services/api/base.service.ts:104]: In base.service we have only the common methods : get all, get by id, remove, add, list. We never add specific methods here. If we have a specific method to add, we add it in the specific service. In your case, in the sample service. But you shouldn't do this now. You should use the same getById method that is beeing used for the other enteties. So please talk to Marius and Lau to create a ticket for the backend and to make the sample getById work like the rest

Comment (sandorbanvitagroup) [reply src/app/_services/api/base.service.ts:104]: It will be done when getAll will be available. It's not working from subComponent services.

**Review by danielrus — DISMISSED** [review]

**Review by adriandumavitagroup — DISMISSED** [review]

**Review by adriandumavitagroup — APPROVED** [review]

**Pull request merged**